dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,364 @@
+
+        if(results == null || results.size()==0){
+            return ret;
+        }
+
+        for (Map<String, String> map : results) {
+            Constructor<?> ctor = classToUse.getConstructor();
+            T object = (T) ctor.newInstance();
+
+            properties = map.keySet().stream().collect(Collectors
+                    .toMap(key -> CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, key), key ->map.get(key)));
+
+            for (String property: properties.keySet()){
+                if (properties.get(property) != null){
+                    if (isFieldPresent(classToUse, String.class, property)){
+                        PropertyUtils.setProperty(object, property, properties.get(property));
+                    }else if (isFieldPresent(classToUse, Integer.TYPE, property)){
+                        PropertyUtils.setProperty(object, property, Integer.parseInt(properties.get(property)));
+                    }else if (isFieldPresent(classToUse, Boolean.TYPE, property)){
+                        PropertyUtils.setProperty(object, property, Boolean.parseBoolean(properties.get(property)));
+                    }else if (isFieldPresent(classToUse, Date.class, property)){
+                        PropertyUtils.setProperty(object, property, df.parse(properties.get(property)));
+                    }else{
+                        Logger.warn(classToUse, "Property " + property + "not set for " + classToUse.getName());
+                    }
+                }
+            }
+
+            ret.add(object);
+        }
+        return ret;
+    }
+
+    /**
+     * Searches a property recursively through classes and superclasses
+     * @param classToUse
+     * @param fieldType
+     * @param property
+     * @return
+     * @throws NoSuchFieldException
+     */
+    private static boolean isFieldPresent(Class classToUse, Class fieldType, String property)
+            throws NoSuchFieldException {
+
+        try{
+            return classToUse.getDeclaredField(property).getType() == fieldType;
+        }catch(NoSuchFieldException e){
+            if (classToUse.getSuperclass()!=null) {
+                return isFieldPresent(classToUse.getSuperclass(), fieldType, property);
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param results
+     * @return
+     */
+    public static List<Link> convertDotConnectMapToLink(List<Map<String,String>> results)
+            throws ParseException {
+
+        List<Link> ret;
+
+        ret = new ArrayList<>();
+
+        if(results != null && !results.isEmpty()){
+            for (Map<String, String> map : results) {
+                Link link = new Link();
+                link.setInode(map.get("inode"));
+
+                if (map.get("show_on_menu") != null && !map.get("show_on_menu").isEmpty()){
+                    link.setShowOnMenu(Boolean.parseBoolean(map.get("show_on_menu")));
+                }
+
+                link.setTitle(map.get("title"));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "title" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)